src/transformers/models/instructblip/modeling_instructblip.py

```diff
@@ -281,6 +281,13 @@ class InstructBlipPreTrainedModel(PreTrainedModel):
         r"language_model.decoder.embed_tokens.weight",
         r"language_model.lm_head.weight",
     ]
+    _keep_in_fp32_modules = ["wo"]
+    _no_split_modules = [
+        "InstructBlipAttention",
+        "T5Block",
+        "OPTDecoderLayer",
+        "InstructBlipQFormerMultiHeadAttention",
+    ]
```

Comment:
I think this should be adapted dynamically depending on the models that are actually loaded, to be more future proof.

Reply:
Sure, makes sense